File: php/features.php
<?php
declare(strict_types=1);

// HarryPotter SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class HarryPotterFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new HarryPotterBaseFeature();
            case "test":
                return new HarryPotterTestFeature();
            default:
                return new HarryPotterBaseFeature();
        }
    }
}

File: py/features.py
# HarryPotter SDK feature factory

from feature.base_feature import HarryPotterBaseFeature
from feature.test_feature import HarryPotterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HarryPotterBaseFeature(),
        "test": lambda: HarryPotterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
